htlcswitch/link.go

```diff
@@ -1738,6 +1750,14 @@ func (l *channelLink) handleDownstreamPkt(pkt *htlcPacket) {
 		_ = l.handleDownstreamUpdateAdd(pkt)
 
 	case *lnwire.UpdateFulfillHTLC:
+		if !l.quiescer.canSendUpdates() {
```

Comment:
Hmm, this is one of those "never should happen" scenarios right? I guess can happen given a slim concurrency window where we mark state as flushing, but the items are already in the mailbox. 

Def need to test this out more, but my understanding is that these are un-ack'd, so they'll sit in the mailbox, to eventually be retransmitted once the connection recycles.

Reply:
Yes, but I believe every subsystem needs to be responsible for ensuring its own consistency so it is included for completeness. As we tease these systems apart a bit better, I think we can get more clear about the caller's responsibilities.

This code is a living system and I'd rather have some redundant checks than have someone change something else and break a fundamental guarantee in a place they didn't touch.

Reply:
> so they'll sit in the mailbox, to eventually be retransmitted once the connection recycles.

Would be nice to add a comment about this on `UpdateFulfillHTLC` and `UpdateFailHTLC`.